server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/EventDeadLettersRoutes.java

```diff
@@ -0,0 +1,218 @@
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = true,
+            name = "group",
+            paramType = "path parameter",
+            dataType = "String",
+            defaultValue = "none",
+            value = "Compulsory. Needs to be a valid group name"),
+        @ApiImplicitParam(
+            required = true,
+            name = "eventId",
+            paramType = "path parameter",
+            dataType = "String",
+            defaultValue = "none",
+            value = "Compulsory. Needs to be a valid eventId")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.OK_200, message = "OK - returns an event detail", response = Event.class),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid group name or event id"),
+        @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "No message with this eventId"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
+    })
+    private String getEventDetails(Request request, Response response) {
+        Group group = parseGroup(request);
+        Event.EventId eventId = parseEventId(request);
+
+        return deadLetters.failedEvent(group, eventId)
+            .map(eventSerializer::toJson)
+            .block();
+    }
+
+    @DELETE
+    @Path("/groups/" + GROUP_PARAM + "/events/" + EVENT_ID_PARAM)
+    @ApiOperation(value = "Deletes an event")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = true,
+            name = "group",
+            paramType = "path parameter",
+            dataType = "String",
+            defaultValue = "none",
+            value = "Compulsory. Needs to be a valid group name"),
+        @ApiImplicitParam(
+            required = true,
+            name = "eventId",
+            paramType = "path parameter",
+            dataType = "String",
+            defaultValue = "none",
+            value = "Compulsory. Needs to be a valid eventId")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "OK - Event deleted"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid group name or event id"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
+    })
+    private Response deleteEvent(Request request, Response response) {
+        Group group = parseGroup(request);
+        Event.EventId eventId = parseEventId(request);
+
+        deadLetters.remove(group, eventId).block();
+        response.status(HttpStatus.NO_CONTENT_204);
+        return response;
+    }
+
+    private Group parseGroup(Request request) {
+        String groupAsString = request.params(GROUP_PARAM);
+        try {
+            return Group.deserialize(groupAsString);
+        } catch (Group.GroupDeserializationException e) {
```

Comment:
After reflexion, still the same. Do we take as granted that a non existing group (which raises the exception on the deserialization) returns 400? Should modify the description in the ticket then, as we have a 404 defined in it for group not registered

Reply:
Well I do think we don't need to stick on the description of the ticket for the following reason:

 - A "not found group" means the FQDN specified is not valid. 
 - A group is valid will automatically "be found" and will never return a 404
 - So, the root cause of a possible 404 is not that a resopurce do not exist, but that the resources asked for **is not valid.

I really don't think a 404 describes this better. A 400 do.